### PR TITLE
Create a Wazuh DB method to set the not synced agents in a Master node

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#5693|

## Description
In this PR a new command is created for **wdb_parse()** to insert all agent information and labels into Global DB

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [ ] Review logs syntax and correct language

- Memory tests for Linux
  - [ ] Valgrind (memcheck and descriptor leaks check)

- [ ] Working on cluster environments
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

### CODE REVIEW

File: src/wazuh_db/wdb_parser.c [hunk 1]
```diff
@@ -3946,3 +3954,85 @@ int wdb_parse_global_set_agent_labels(wdb_t * wdb, char * input, char * output)
...
+    root = cJSON_ParseWithOpts(input, &error, TRUE);
+    if (!root) {
+        mdebug1("Global DB Invalid JSON syntax updating unsynced agents.");
+        mdebug2("Global DB JSON error near: %s", error);
+        snprintf(output, OS_MAXSTR + 1, "err Invalid JSON syntax, near '%.32s'", input);
+        return OS_INVALID;
+
+    } else {
+        /*
+        * Iterating over the array with cJSON_ArrayForEach()
```
Comment (DProvinciani) [inline src/wazuh_db/wdb_parser.c:3985]: Not sure whether this comment and explanation are necessary. This function is already used in many parts of the Wazuh code.
Different from this is the comment above where you are making other developers be careful of changing the `cJSON_ParseWithOpts` function that is thread-safe and is well used here where we are in a function executed by many threads.

File: src/wazuh_db/wdb_global.c [hunk 2]
```diff
@@ -110,3 +134,68 @@ int wdb_global_set_agent_label(wdb_t *wdb, int id, char* key, char* value) {
...
+            }
+       
+        } else if (cJSON_IsString(json_field) && json_field->valuestring != NULL) {
+            if (sqlite3_bind_text(stmt, n + 1 , json_field->valuestring, -1, NULL) != SQLITE_OK) {
+                merror("DB(%s) sqlite3_bind_text(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+                return OS_INVALID;
+            }
+        
+        } else{
+            if (sqlite3_bind_null(stmt, n+1) != SQLITE_OK) {
```
Comment (DProvinciani) [inline src/wazuh_db/wdb_global.c:172]: Just by curious... Is this to handle the case where for an agent, one of the fields does not contain information?

File: src/wazuh_db/wdb_global.c [hunk 3]
```diff
@@ -110,3 +134,68 @@ int wdb_global_set_agent_label(wdb_t *wdb, int id, char* key, char* value) {
...
+            }
+        }
+    }
+
+    if (sqlite3_bind_int(stmt, n+1, WDB_SYNCED) != SQLITE_OK) {
+        merror("DB(%s) sqlite3_bind_int(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+        return OS_INVALID;
+    }
+
+    json_field = cJSON_GetObjectItemCaseSensitive(json_agent, "id");
```
Comment (DProvinciani) [inline src/wazuh_db/wdb_global.c:184]: Why not manage the `id` also in the `for` loop?

File: src/wazuh_db/wdb_parser.c (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_parser.c:3985]: OK, thank you!

File: src/wazuh_db/wdb_parser.c [hunk 4]
```diff
@@ -3946,3 +3954,85 @@ int wdb_parse_global_set_agent_labels(wdb_t * wdb, char * input, char * output)
...
+            }
+            // Checking for labels
+            json_labels = cJSON_GetObjectItemCaseSensitive(json_agent, "labels");
+            if(cJSON_IsArray(json_labels)){
+                // The JSON has a label array
+                // Removing old labels from the labels table before inserting
+                json_field = cJSON_GetObjectItemCaseSensitive(json_agent, "id");
+                agent_id = cJSON_IsNumber(json_field) ? json_field->valueint : -1; 
+
+                if (OS_SUCCESS != wdb_global_del_agent_labels(wdb, agent_id)) {
```
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_db/wdb_parser.c:4004]: Thank you, added!

File: src/wazuh_db/wdb_parser.c [hunk 5]
```diff
@@ -3946,3 +3954,85 @@ int wdb_parse_global_set_agent_labels(wdb_t * wdb, char * input, char * output)
...
+                    cJSON_Delete(root);
+                    return OS_INVALID;
+                }
+                // For every label in array, insert it in the database
+                cJSON_ArrayForEach(json_label, json_labels){
+                    json_key = cJSON_GetObjectItemCaseSensitive(json_label, "key");
+                    json_value = cJSON_GetObjectItemCaseSensitive(json_label, "value");
+                    json_id = cJSON_GetObjectItemCaseSensitive(json_label, "id");
+
+                    if(cJSON_IsString(json_key) && json_key->valuestring != NULL){
```
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_db/wdb_parser.c:4016]: Replaced!

File: src/wazuh_db/wdb_global.c (hunk 2 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_global.c:172]: This is unnecessary, removed!

File: src/wazuh_db/wdb_global.c (hunk 3 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_global.c:184]: The `id` is now in the loop.

**Pull request merged**